File: client/modules/native-settings/ios/NativeSettingsModule.swift
import ExpoModulesCore

public final class NativeSettingsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("NativeSettings")

    View(NativeSettingsListView.self) {
      ViewName("NativeSettingsListView")

      Prop("screenId") { (view: NativeSettingsListView, screenId: String?) in
        view.updateScreenId(screenId)
      }

      Prop("sectionsJson") { (view: NativeSettingsListView, sectionsJson: String?) in
        view.updateSectionsJson(sectionsJson ?? "[]")
      }

      Events(
        "onPressItem",
        "onToggleChange",
        "onMenuAction",
        "onNavigate",
        "onExpandChange",
        "onError"
      )
    }

    View(NativeSelectionListView.self) {
      ViewName("NativeSelectionListView")

      Prop("screenId") { (view: NativeSelectionListView, screenId: String?) in
        view.updateScreenId(screenId)
      }

      Prop("payloadJson") { (view: NativeSelectionListView, payloadJson: String?) in
        view.updatePayloadJson(payloadJson ?? "{}")
      }

      Events(
        "onPressItem",
        "onSelectionCommit",
        "onError"
      )
    }

    View(NativeCategoryManagerView.self) {
      ViewName("NativeCategoryManagerView")

      Prop("screenId") { (view: NativeCategoryManagerView, screenId: String?) in
        view.updateScreenId(screenId)
      }

      Prop("sectionsJson") { (view: NativeCategoryManagerView, sectionsJson: String?) in
        view.updateSectionsJson(sectionsJson ?? "[]")
      }

      Events(
        "onPressItem",
        "onMenuAction",
        "onReorderCommit",
        "onSwipeAction",
        "onRequestDelete",
        "onError"
      )
    }

    View(NativePickerHostView.self) {
      ViewName("NativePickerHostView")

      Prop("screenId") { (view: NativePickerHostView, screenId: String?) in
        view.updateScreenId(screenId)
      }

      Prop("payloadJson") { (view: NativePickerHostView, payloadJson: String?) in
        view.updatePayloadJson(payloadJson ?? "{}")
      }

      Events("onError")
    }
  }
}

File: client/modules/native-settings/android/src/main/java/expo/modules/nativesettings/NativeSettingsModule.kt
package expo.modules.nativesettings

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NativeSettingsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("NativeSettings")

    View(NativeSettingsListView::class) {
      Prop("screenId") { view: NativeSettingsListView, screenId: String? ->
        view.updateScreenId(screenId)
      }

      Prop("sectionsJson") { view: NativeSettingsListView, sectionsJson: String? ->
        view.updateSectionsJson(sectionsJson ?: "[]")
      }

      Events(
        "onPressItem",
        "onToggleChange",
        "onMenuAction",
        "onNavigate",
        "onExpandChange",
        "onError"
      )
    }

    View(NativeSelectionListView::class) {
      Prop("screenId") { view: NativeSelectionListView, screenId: String? ->
        view.updateScreenId(screenId)
      }

      Prop("payloadJson") { view: NativeSelectionListView, payloadJson: String? ->
        view.updatePayloadJson(payloadJson ?: "{}")
      }

      Events(
        "onPressItem",
        "onSelectionCommit",
        "onError"
      )
    }

    View(NativeCategoryManagerView::class) {
      Prop("screenId") { view: NativeCategoryManagerView, screenId: String? ->
        view.updateScreenId(screenId)
      }

      Prop("sectionsJson") { view: NativeCategoryManagerView, sectionsJson: String? ->
        view.updateSectionsJson(sectionsJson ?: "[]")
      }

      Events(
        "onPressItem",
        "onMenuAction",
        "onReorderCommit",
        "onSwipeAction",
        "onRequestDelete",
        "onError"
      )
    }

    View(NativePickerHostView::class) {
      Prop("screenId") { view: NativePickerHostView, screenId: String? ->
        view.updateScreenId(screenId)
      }

      Prop("payloadJson") { view: NativePickerHostView, payloadJson: String? ->
        view.updatePayloadJson(payloadJson ?: "{}")
      }

      Events("onError")
    }
  }
}
